drivers/usb/device/usb_dw_registers.h

```diff
@@ -132,8 +136,41 @@ BUILD_ASSERT(sizeof(struct usb_dw_reg) <= 0x0D00);
 #define USB_DW_GRXSTSR_PKT_CNT_OFFSET		4
 #define USB_DW_GRXSTSR_EP_NUM_MASK		(0xF << 0)
 
-/* ? register, offset: 0x0050 */
-#define USB_DW_HWCFG4_DEDFIFOMODE		BIT(25)
+/* Application (vendor) general purpose registers, offset: 0x0038 */
+#define USB_DW_GGPIO_STM32_VBDEN		BIT(21)
+#define USB_DW_GGPIO_STM32_PWRDWN		BIT(16)
+
+/* GHWCFG1 register, offset: 0x0044 */
+#define USB_DW_GHWCFG1_EPDIR_MASK(i)		(0x3 << (i << 2))
+#define USB_DW_GHWCFG1_EPDIR_SHIFT(i)		(i << 2)
```

Comment:
The calculation is incorrect, e.g. EP 15 direction is on bits [31:30], but `(0x3 << (15 << 2))` is way outside 32-bit integer.
```suggestion
#define USB_DW_GHWCFG1_EPDIR_MASK(i)		(0x3 << (i * 2))
#define USB_DW_GHWCFG1_EPDIR_SHIFT(i)		(i * 2)
```

While SHIFT makes more sense, the convention in this header seems to be OFFSET instead of SHIFT.